modules/core/src/main/java/org/apache/ignite/internal/managers/systemview/walker/DataRegionPagesListViewWalker.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.managers.systemview.walker;
+
+import org.apache.ignite.internal.processors.cache.persistence.freelist.PagesListView;
+import org.apache.ignite.spi.systemview.view.SystemViewRowAttributeWalker;
+
+/**
+ * {@link PagesListView} attributes walker for data regions.
+ * 
+ * @see PagesListView
+ */
+public class DataRegionPagesListViewWalker implements SystemViewRowAttributeWalker<PagesListView> {
```

Comment:
We should avoid several walkers for a same row class. Let's create two separate class for new system views of use single walker.

Reply:
Why not? We avoid code duplication when use 2 walkers for 1 view class. 
In case of two different views we also should expose two different methods from free-lists or expose one method with some parameters class and do some unclear class casting.

Reply:
Because it's how system view designed -  the walker should be auto-generated.
I can't see a reason to break it.